analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/RestrictDeserializedTypesBase.cs

```diff
@@ -26,4 +28,166 @@ public abstract class RestrictDeserializedTypesBase : SymbolicRuleCheck
+        }
+        else if (objectCreation.Type.Is(LosFormatter) && !EnableMacIsTrue(state, objectCreation))
+        {
+            ReportIssue(objectCreation.WrappedOperation, VerifyMacMessage);
+        }
+        return state;
+    }
+
+    private bool UnsafeJavaScriptSerializer(ProgramState state, IObjectCreationOperationWrapper objectCreation, out SyntaxNode resolveTypeDeclaration)
+    {
+        resolveTypeDeclaration = null;
+        return objectCreation.Type.Is(JavaScriptSerializer)
+            && objectCreation.Arguments.Length == 1
+            && UnsafeResolver(state, objectCreation.Arguments[0].ToArgument().Value, out resolveTypeDeclaration);
+    }
+
+    private bool UnsafeResolver(ProgramState state, IOperation operation, out SyntaxNode resolveTypeDeclaration)
+    {
+        resolveTypeDeclaration = null;
+        operation = state.ResolveCaptureAndUnwrapConversion(operation);
+        if (operation.Type.Is(KnownType.System_Web_Script_Serialization_SimpleTypeResolver))
+        {
+            return true;
+        }
+        else if (DeclarationCandidates(operation)?.FirstOrDefault(IsResolveTypeMethod) is { } declaration
+            && !ThrowsOrReturnsNull(declaration))
+        {
+            resolveTypeDeclaration = declaration;
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
+
+    private static bool EnableMacIsTrue(ProgramState state, IObjectCreationOperationWrapper objectCreation) =>
+        objectCreation.ArgumentValue("enableMac") is { } enableMacArgument
+        && state[enableMacArgument]?.HasConstraint(BoolConstraint.True) is true;
+
+    private ProgramState ProcessBinderAssignment(ProgramState state, IAssignmentOperationWrapper assignment)
+    {
+        if (BinderAssignmentInstance(state, assignment) is { } instance)
+        {
+            var constraint = BinderIsSafe(state, assignment, out var bindToTypeDeclaration)
+                ? SerializationConstraint.Safe
+                : SerializationConstraint.Unsafe;
+            if (constraint == SerializationConstraint.Unsafe)
+            {
+                additionalLocationsForOperations[instance] = bindToTypeDeclaration;
+            }
+            state = state.SetOperationConstraint(instance, constraint);
+
+            if (instance.TrackedSymbol() is { } symbol)
+            {
+                if (constraint == SerializationConstraint.Unsafe)
+                {
+                    additionalLocationsForSymbols[symbol] = bindToTypeDeclaration;
+                }
+                state = state.SetSymbolConstraint(symbol, constraint);
+            }
+            return state;
+        }
+        return null;
+    }
+
+    private static IOperation BinderAssignmentInstance(ProgramState state, IAssignmentOperationWrapper assignment) =>
+        state.ResolveCaptureAndUnwrapConversion(assignment.Target).AsPropertyReference() is { Property.Name: "Binder", Instance: { } propertyInstance }
```

Comment:
```suggestion
        state.ResolveCaptureAndUnwrapConversion(assignment.Target).AsPropertyReference() is { Property.Name: nameof(IFormatter.Binder), Instance: { } propertyInstance }
```

The same for other occurrences, like "Deserialize", "BindToType" and "ResolveType".

Reply:
I don't think I can do it for "ResolveType" because we don't have access to `System.Web.Script.Serialization.JavaScriptTypeResolver`.